sdk/samples/iot/hub/src/paho_iot_hub_pnp_example.c

```diff
@@ -0,0 +1,744 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifdef _MSC_VER
+// warning C4201: nonstandard extension used: nameless struct/union
+#pragma warning(disable : 4201)
+#endif
+#include <paho-mqtt/MQTTClient.h>
+#ifdef _MSC_VER
+#pragma warning(default : 4201)
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+#ifdef _WIN32
+// Required for Sleep(DWORD)
+#include <Windows.h>
+#else
+// Required for sleep(unsigned int)
+#include <unistd.h>
+#endif
+
+#include <azure/core/az_json.h>
+#include <azure/core/az_result.h>
+#include <azure/core/az_span.h>
+#include <azure/iot/az_iot_hub_client.h>
+
+#ifdef _MSC_VER
+// "'getenv': This function or variable may be unsafe. Consider using _dupenv_s instead."
+#pragma warning(disable : 4996)
+#endif
+
+// DO NOT MODIFY: Device ID Environment Variable Name
+#define ENV_DEVICE_ID "AZ_IOT_DEVICE_ID"
```

Comment:
Informational for @ericmitt - note that the embedded won't follow the same conventions you wanted for other SDK variable names.  This is unavoidable because embedded doesn't have connection strings but needs pre-split up values, but you should track.

Reply:
Got it, Thanks John